src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs

```diff
@@ -62,7 +82,7 @@ public Task Stop()
             if (!uint256.TryParse(txid, out trxid))
                 throw new ArgumentException(nameof(txid));
 
-            Transaction trx = await this.FullNode.NodeService<IPooledTransaction>(true)?.GetTransaction(trxid);
+            Transaction trx = await this.pooledTransaction.GetTransaction(trxid);
```

Comment:
Can the service injected be potentially null? The original code seemed to think so but not sure if that's a valid assumption or not.

Reply:
It's valid. If the default value == null then DI wont fail on the missing components and just use null. 

But there was a wrong asusmption of another type: 

`unspentOutputs = await this.FullNode.NodeService<IPooledGetUnspentTransaction>()?.GetUnspentTransactionAsync(trxid);`

'?' doesn't make any sense here since if there is no service- it will throw, not return null.

Reply:
For the parts in the fullnodecontroller where we expect an exception to be thrown when getting the service and it is not available the .? does not make any sense.